### ISSUE TITLE
Final testing for release

### ISSUE THREAD

**Issue by miklasr** [ESE-Peasy/PosturePerfection#145]
# Goal

Conduct final testing of the full software before making the first full release. Each team member should attempt to build from source **completely from scratch**. The following plan should help everyone through the full testing:

# Plan

- Flash the Raspberry Pi with a fresh Raspberry Pi OS image (ensures there are no downloaded dependencies we have forgotten about)
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR
- Run the `./scripts/install_dependencies.sh` script
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR
- Run the `./scripts/build -enable-testing` script
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR
- Build and start the notification receiver
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR
- Start the application and make sure all buttons and sliders work, logo shows up, notifications are sent, etc.
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR

---

- Flash the RPi with a fresh image again
  - [x] CB
  - [ ] AM
  - [ ] AR
  - [ ] MR
- Install using the quick-install method
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR
- Test the full system again
  - [x] CB
  - [x] AM
  - [x] AR
  - [x] MR

**Comment by C-Begley** [comment]
Built successfully following steps and prompts from readme and install scripts

=================================================================================================
Previous attempts
Some commands to add to setup pi install
Maybe give a link to the raspberry pi setup page?

1. Setup Pi Camera
```Sudo raspi-config```
Interface options -> Enable Camera

1. Install CMake using 
```pip3 install cmake```

1. Add pip to install path
```PATH=$PATH:/home/pi/.local/bin```

1. Clone Repo
```git clone https://github.com/ESE-Peasy/PosturePerfection.git```

1. Navigate to Repo
```cd PosturePerfection```

1. Install dependencies
```./scripts/install_dependencies```

Couldn't instal dependencies, git was hanging
Switching to new branch (#148)) and PR (#149) and starting again

1. Setup Pi Camera
```Sudo raspi-config```
Interface options -> Enable Camera

1. Clone Repo
```git clone https://github.com/ESE-Peasy/PosturePerfection.git```

1. Navigate to Repo
```cd PosturePerfection```

1. Install dependencies
```./scripts/install_dependencies```

1. Received prompt to Install CMake using 
```pip3 install cmake```

1. Add pip to install path
```PATH=$PATH:/home/pi/.local/bin```

1. Flashed and got new update to get additional warning about CMake

**Comment by miklasr** [comment]
My steps for source build:
```
# Download source
git clone https://github.com/ESE-Peasy/PosturePerfection.git

# Install CMake
pip3 install cmake

# Add pip install path to PATH
PATH=$PATH:/home/pi/.local/bin

# Install dependencies
./scripts/install_dependencies.sh

# Install cpplint
pip3 install cpplint

# Build with tests
./scripts/build.sh -enable-testing
```

**Comments/remarks**

- Change the comment for CMake not installed to suggest `pip3 install cmake`. Don't run the command though as users may want to install CMake differently (as CMake is a fairly core dependency for many projects).
- Should we add `pip3 install cpplint` to install dependencies?

**Comment by Ashwin-MJ** [comment]
I have a `cpplint` error for `./scripts/build.sh -enable-testing`:

```
src/gui/mainwindow.cpp:256:  Lines should be <= 80 characters long  [whitespace/line_length]
```

Update: Fixed in #149

**Comment by C-Begley** [comment]
Installed from source and the quick install on the most recent changes. Flashed before hand both times. Everything appears to work
